### ISSUE TITLE
[REACT] Crash when using "custom" locale without explicit `startDate` and `endDate`

### ISSUE THREAD

**Issue by ghulamghousdev** [bryntum/support#10465]
[Forum post](https://forum.bryntum.com/viewtopic.php?f=51&t=31386&p=159739#p159739)

Hi,
I'm using schedulerpro in the version 6.1.2 and would like to update to 6.1.3 but the scheduler crashes with the message: "Invalid time axis configuration or filter, please check your input data." when using it with a non default locale without providing startDate and endDate.

Repro: https://github.com/ChristopherSchoening/bryntum-scheduler-pro-bug-repro-custom-locale

any help is much appreciated.

[Error.zip](https://github.com/user-attachments/files/18079078/Error.zip)
<img width="777" alt="Screenshot 2024-12-10 at 5 02 10 PM" src="https://github.com/user-attachments/assets/42dd4a42-17ed-4278-94ed-d1832debedab">

**Comment by ChristopherSchoening** [comment]
the issue is also reproducable in v6.1.4

**Comment by ghulamghousdev** [comment]
> the issue is also reproducable in v6.1.4

Yup. I know the issue and already implementing a fix. Working on a test at the moment.

**Comment by ghulamghousdev** [comment]
@mx-bernhard Though I had added test and everything works in the PR, but I have a question. What is your use-case with passing an empty locale? 

Because when the locale without `unitNames` and `unitAbbrevations` is being set, we call a method `setupDurationRegEx` from `applyLocale` in `dateHelper.js` to construct a `regex`. In our case, as `units` are not preset, it is being resulted in the below: `/^(?:)$/`
And when `changeRange` method in `schedulerBase` is executed, we parse the `timeUnit` but as the above `regex` does not have any `unit` in it, it returns `null`
and `parseDuration` also returns `null` as `regex` does not have any unit. That is why range is being set as `null` in scheduler and parsing fails. 

But all said, can you please share your use-case?
